kubernetes/autosharding/kube-state-metrics-cluster-role.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
The autosharding manifests use a statefulset instead of deployment to allow the identification of a nominal within a statefulset (as in the "self replica" out of the "total replicas"). As  well as the permissions for the pods to discover their owning statefulset.

Reply:
Let me know if it still needs to be clarified elsewhere. I added a comment to the readme referencing the manifests.

Reply:
Discussed out of band: we separated the "regular" deployment option and autosharding into separate directories.